api/spec/scanner-adapter-openapi-v1.1.yaml

```diff
@@ -0,0 +1,432 @@
+openapi: 3.0.0
+info:
+  title: Harbor Scanner Adapter API
+  description: |
+    ## Overview
+
+    This API must be implemented in order to register a new artifact scanner in [Harbor](https://goharbor.io) registry.
+
+    The [/scan](#operation/AcceptScanRequest) and [/scan/{scan_request_id}/report](#operation/GetScanReport) operations are responsible for the actual scanning and return a scan report that is visible in the Harbor web console.
+
+    The [/scan](#operation/AcceptScanRequest) operation is asynchronous. It should enqueue the job for processing a scan request and return the identifier. This allows Harbor to poll a corresponding scan report with the
+    [/scan/{scan_request_id}/report](#operation/GetScanReport) operation. Harbor will call the
+    [/scan/{scan_request_id}/report](#operation/GetScanReport) operation periodically periodically until it returns 200 or 500 status codes.
+
+    The [/metadata](#operation/GetMetadata) operation allows a Harbor admin to configure and register a scanner
+    and discover its capabilities.
+
+    ## Supported consumed MIME types
+
+    - `application/vnd.oci.image.manifest.v1+json`
+    - `application/vnd.docker.distribution.manifest.v2+json`
+
+    ## Supported produced MIME types
+
+    - `application/vnd.security.vulnerability.report; version=1.1`
```

Comment:
If we want to introduce a generic mime type here, we have to update all corresponding examples and refs in this spec. Currently we use both new name and the old one.

Reply:
Done. Corrected the mime type with the `scan/{scan_request_id}/report` API definition.

Reply:
@prahaladdarkin I think that to keep this API change backward compatible we should not replace `application/vnd.scanner.adapter.vuln.report.harbor+json; version=1.0` but add `application/vnd.security.vulnerability.report; version=1.1` to the list of supported produce MIME types. Then at some point we can deprecate `application/vnd.scanner.adapter.vuln.report.harbor+json; version=1.0` once we make sure that every scanner adapter has migrated. WDYT?

Reply:
Done. Re-added the older MIME type to begin the process of deprecation

Reply:
Done. Corrected to perform a deprecation rather than removal of the existing MIME type